optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/manager/DefaultSolverManager.java

```diff
@@ -142,43 +152,97 @@ public void solve(
             }
             newSolverTask = new SolverTask<>(problemId, solverFactory.buildSolver(), planningProblem);
             problemIdToSolverTaskMap.put(problemId, newSolverTask);
+            problemIdToEventHandlerFuturesMap.put(problemId, Collections.synchronizedList(new ArrayList<>()));
```

Comment:
Is there any particular reason for using `Collections.synchronizedList()`? Java has concurrent collections whose performance is generally better.

If you insist on using synchronized collections, please make sure you comply [with its Javadoc](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#synchronizedList(java.util.List)).

Reply:
Thanks for pointing that out! No particular reason, actually `CopyOnWriteArrayList` is better than implementing the synchronization manually while iterating over the list.